### PR TITLE
Add notifications to status bar and do a final ack to clear the lock

### PR DESCRIPTION
**Pull request by tadeoos**

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -280,10 +288,31 @@ export default class ReadwisePlugin extends Plugin {
...
+        `${baseURL}/api/obsidian/sync_ack`,
+        {
+          headers: {...this.getAuthHeaders(), 'Content-Type': 'application/json'},
+          method: "POST",
+        })
+    } catch (e) {
+      console.log("Readwise Official plugin: fetch failed to acknowledged sync: ", e);
+    }
+    if (response && response.ok) {
+
```
Comment (adam-lynch) [inline src/main.ts:309]: Empty

File: src/status.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      this.lastMessageTimestamp = Date.now();
+    } else if (this.currentMessage) {
+      let messageAge = Date.now() - this.lastMessageTimestamp;
+      if (messageAge >= this.currentMessage.timeout) {
+        this.currentMessage = null;
+        this.lastMessageTimestamp = null;
+      }
+    } else {
+      this.statusBarEl.setText("");
+    }
```
Comment (adam-lynch) [inline src/status.ts:45]: I'd find this clearer

```suggestion
    if(this.currentMessage) {
      let messageAge = Date.now() - this.lastMessageTimestamp;
      if (messageAge >= this.currentMessage.timeout) {
        this.currentMessage = null;
        this.lastMessageTimestamp = null;
      }
    }
    else if(this.messages.length) {
      this.currentMessage = this.messages.shift();
      this.statusBarEl.setText(this.currentMessage.message);
      this.lastMessageTimestamp = Date.now();
      return;
    }
    else {
      this.statusBarEl.setText("");
    }
```

That's just re-arranging the `if`s

**Review by adam-lynch — APPROVED** [review]

File: src/main.ts [hunk 3]
```diff
@@ -280,10 +288,31 @@ export default class ReadwisePlugin extends Plugin {
     }
     // close the ZipReader
     await zipReader.close();
+    await this.acknowledgeSyncCompleted(buttonContext)
```
Comment (TristanH) [inline src/main.ts:291]: What happens if we have some error in the javascript above (or eg the user quits Obsidian) while syncing, and this never gets called?

The client will just not sync for the next hour, right? (until lock on our server expires?)

File: src/main.ts (hunk 1 above)
Comment (tadeoos) [reply src/main.ts:309]: dank

File: src/status.ts (hunk 2 above)
Comment (tadeoos) [reply src/status.ts:45]: 👍🏻

File: src/main.ts (hunk 3 above)
Comment (TristanH) [reply src/main.ts:294]: @tadeoos ok, but is this necessary though? would it not be possible for the server to clear the lock, so this client try again without waiting a whole hour?

**Review by TristanH — APPROVED** [review]

File: src/main.ts (hunk 3 above)
Comment (tadeoos) [reply src/main.ts:292]: yeah, I changed the behavior to just continue if we have a lock from the same client already

**Pull request merged**